src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -120,7 +156,33 @@ public DosFileOperationResult CreateFileUsingHandle(string fileName, ushort file
             testFileStream?.Dispose();
         }
 
-        return OpenFileInternal(fileName, prefixedPath, "rw");
+        return OpenFileInternal(fileName, prefixedPath, FileAccessMode.ReadWrite);
+    }
+
+    /// <summary>
+    /// Gets another file handle for the same file
+    /// </summary>
+    /// <param name="fileHandle">The handle to a file.</param>
+    /// <param name="newHandle">The new handle to a file.</param>
+    /// <returns>A <see cref="DosFileOperationResult"/> with details about the result of the operation.</returns>
+    public DosFileOperationResult ForceDuplicateFileHandle(ushort fileHandle, ushort newHandle) {
+        if (fileHandle == newHandle) {
+            return DosFileOperationResult.Error(ErrorCode.InvalidHandle);
+        }
+        if (!IsValidFileHandle(newHandle)) {
+            return DosFileOperationResult.Error(ErrorCode.InvalidHandle);
+        }
+        if (OpenFiles[newHandle] != null) {
+            return DosFileOperationResult.Error(ErrorCode.InvalidHandle);
+        }
+        if (GetOpenFile(fileHandle) is not VirtualFileBase file) {
```

Comment:
is not null? VirtualFileBase? is the return type of the method

Reply:
Not very clear, I will replace it.